sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1191,6 +1191,30 @@ class Dataset[T] private[sql](
       valueColumnName: String): DataFrame =
     unpivot(ids.toArray, variableColumnName, valueColumnName)
 
+  /** @inheritdoc */
+  def transpose(indexColumn: Column): DataFrame = withPlan {
+    if (this.columns.isEmpty) {
+      LocalRelation()
+    } else {
+      UnresolvedTranspose(
+        indexColumn.named,
+        logicalPlan
+      )
+    }
+  }
+
+  /** @inheritdoc */
+  def transpose(): DataFrame = withPlan {
```

Comment:
may I ask why not let `ResolveTranspose` handle `this.columns.isEmpty` and `this.columns.head`?

Reply:
Good point! Moving to ResolveTranspose benefits Spark Connect and provides a more readable plan.
Would you please review [b9a0c80](https://github.com/apache/spark/pull/47884/commits/b9a0c80d47680c54db50369a25e90b56f8f6333c) for the change?